torch/distributions/distribution.py

```diff
@@ -29,6 +29,23 @@ def event_shape(self):
         """
         return self._event_shape
 
+    @property
+    def params(self):
+        """
+        Returns a dictionary from param names to `Constraint` objects that
+        should be satisfied by each parameter of this distribution. For
+        distributions with multiple parameterization, only one complete
+        set of parameters should be specified in `.params`.
+        """
+        raise NotImplementedError
+
+    @property
+    def support(self):
```

Comment:
Hmm I think that if we want to make these attrs be part of the class interface too this won't be enough - `Distribution.params` will still be a valid thing and return the descriptor. If we want to make it raise an error on all kinds of accesses we would need to make it a descriptor that always raises in `__get__`

Not a big deal ofc, I'm ok with merging this as is. Just pointing out a possible improvement. Let me know what you want to do.

Reply:
Sounds reasonable, but I'm not sure how to implement it without a nasty `__getattr__` override. I'm happy merging now and thinking about how to do this.